File: config/phoenix/configuration.js
const MARGIN = 10;

const WORKSPACES = [
  {
    workspace: "Browser",
    profile: "Full",
    layout: "Fullscreen",
    applications: ["Arc"],
  },
  {
    workspace: "Terminal",
    profile: "Full",
    layout: "Fullscreen",
    applications: ["Ghostty"],
  },
  {
    workspace: "Notes",
    profile: "Full",
    layout: "Fullscreen",
    applications: ["Obsidian"],
  },
  {
    workspace: "Calendar",
    profile: "Full",
    layout: "Fullscreen",
    applications: ["BusyCal"],
  },
  {
    workspace: "Music",
    profile: "Full",
    layout: "Fullscreen",
    applications: ["Spotify", "Music"],
  },
  {
    workspace: "Slack",
    profile: "Full",
    layout: "Fullscreen",
    applications: ["Slack"],
  },
  {
    workspace: "Browser",
    profile: "Split",
    layout: "Fullscreen",
    applications: ["Arc"],
  },
  {
    workspace: "Terminal",
    profile: "Split",
    layout: "Main Stack",
    applications: ["Arc", "Ghostty"],
  },
  {
    workspace: "Notes",
    profile: "Split",
    layout: "Main Stack",
    applications: ["Arc", "Obsidian"],
  },
  {
    workspace: "Calendar",
    profile: "Split",
    layout: "Fullscreen",
    applications: ["BusyCal"],
  },
  {
    workspace: "Music",
    profile: "Split",
    layout: "Fullscreen",
    applications: ["Spotify", "Music"],
  },
  {
    workspace: "Slack",
    profile: "Split",
    layout: "Main Stack",
    applications: ["Arc", "Slack"],
  },
];

function getWorkspaceConfiguration(profile, workspace) {
  let workspaceConfiguration = WORKSPACES.find((configuration) => {
    return configuration.workspace === workspace && configuration.profile === profile;
  });

  if (!workspaceConfiguration) {
    throw new Error(`No configuration found for workspace "${workspace}" and profile "${profile}"`);
  }

  return workspaceConfiguration;
}

File: config/window-manager/configuration.rb
require 'yaml'

class Configuration
  WorkspaceConfiguration = Data.define(:profile, :workspace, :layout, :applications) do
    def matches?(profile, workspace)
      self.profile == profile && self.workspace == workspace
    end
  end

  DisplayConfiguration = Data.define(:name, :external, :menu_bar_height, :dock_height) do
    def profile
      external ? 'Split' : 'Full'
    end

    def matches?(name)
      self.name == name
    end
  end

  class << self
    def margin
      config['margin']
    end

    def displays
      @displays ||= config['displays'].map { DisplayConfiguration.new(**_1) }
    end

    def find_display(name)
      display_config = displays.find { _1.matches?(name) }

      unless display_config
        raise "No configuration found for display with name #{name.inspect}"
      end

      display_config
    end

    def workspaces
      @workspaces ||= config['workspaces'].map { WorkspaceConfiguration.new(**_1) }
    end

    def find_workspace(profile, workspace)
      workspace_config = workspaces.find { _1.matches?(profile, workspace) }

      unless workspace_config
        raise "No configuration found for workspace '#{workspace}' and profile '#{profile}'"
      end

      workspace_config
    end

    private

    def config
      @config ||= YAML.load_file(File.join(__dir__, 'configuration.yml'))
    end
  end
end
